app/src/main/java/org/mozilla/fenix/addons/InstalledAddonDetailsFragment.kt

```diff
@@ -211,6 +212,28 @@ class InstalledAddonDetailsFragment : Fragment() {
         }
     }
 
+    private fun bindAllowInPrivateBrowsingSwitch(view: View) {
+        val switch = view.allow_in_private_browsing_switch
+        switch.isChecked = addon.isAllowedInPrivateBrowsing()
+        switch.setOnCheckedChangeListener { v, isChecked ->
+            val addonManager = v.context.components.addonManager
+            switch.isClickable = false
+            view.remove_add_on.isEnabled = false
+            addonManager.setAddonAllowedInPrivateBrowsing(
+                addon,
+                isChecked,
+                onSuccess = {
```

Comment:
Should we also handle `onError` ? As we are disabling the the clicks `switch.isClickable = false`

Reply:
Nice catch, we should reset the UI elements again on any callbacks :).